triton/stream.py

```diff
@@ -234,7 +236,12 @@ def _select_shard_ids(self, shard_nums):
         return shard_ids
 
     def put(self, **kwargs):
-        data = msgpack.packb(kwargs)
+        try:
+            data = msgpack.packb(kwargs)
+        except TypeError:
+            # If we fail to serialize our context, we can try again with an
+            # enhanced packer (it's slower though)
```

Comment:
I couldn't imagine more than a 10% penalty; haven't benchmarked (this comment is from original author). Forcing serialization to users has lead to much confusion and some lost data.